docs: Metrics CLI versioning support analysis (#1480)
Flank/flank-dashboard: metrics/cli/docs/features/supported_storage_version/01_supported_storage_version.md

# Supported Storage Version
> Feature description / User story.

> Improve Metrics Web application deployment process to include the supported storage version passing and storage version saving. 

Reviewer: 🏷️ 
What do you think? To make it easier for a reader
```suggestion
Improve Metrics Web application deployment process to include the supported storage version passing and saving. 
```

## Contents

- [**Analysis**](#analysis)
    - [Feasibility study](#feasibility-study)
    - [Requirements](#requirements)
    - [Landscape](#landscape)
    - [Prototyping](#prototyping)
    - [System modeling](#system-modeling)

# Analysis
> Describe general analysis approach.

During the analysis section, we are going to define the requirements for this feature and choose whether to use the existing approach or develop a custom one to implement this feature.

### Feasibility study
> A preliminary study of the feasibility of implementing this feature.

> Since the `Metrics Web` application would receive updates after the initial release, we should have a convenient way of updating the `Metrics Web` application even if the database has any significant changes. To be able to update the application safely, we are going to introduce a storage version and a supported storage version for all `Metrics` applications. It will allow us to block the application from interaction with the persistent storage in case the application does not support the current storage version or the storage is currently updating. To simplify the `Metrics Web` application deployment process, we should update the `Metrics CLI` to make it set the supported storage version to the `Metrics Web` application.

Reviewer: :readable:
Would it be better to make the section more granular?
```suggestion
Since the `Metrics Web` application would receive updates after the initial release, we should have a convenient way of updating the `Metrics Web` application, even if the database has no significant changes.

To be able to update the application safely, we are going to introduce a storage version and a supported storage version for all `Metrics` applications. This will allow us to block the application from interaction with the persistent storage in case the application does not support the current storage version or the storage is currently updating. 

To simplify the `Metrics Web` application deployment process, we should update the `Metrics CLI` to make it set the supported storage version to the `Metrics Web` application.
```

Since the storing of the supported storage versions is pretty custom, we are going to use the custom solution for setting this version in the `Metrics CLI`. 

### Prototyping
> Create a simple prototype to confirm that implementing this feature is possible.

So, to provide a supported storage version to the `Metrics Web` application during the deployment process, we should follow the next steps:

1. Fetch the storage version from the `STORAGE_VERSION` file under the `metrics` package.
2. Pass the fetched storage version to the `Metrics Web` application environment as a `supported storage version` during the application building process.

Let's consider the code snippets proving that this feature is possible to implement:

> To read the data from the file, we are going to use the [File](https://api.dart.dev/stable/2.13.1/dart-io/File-class.html) class, providing an ability to get the file content using the following code:

Reviewer: :label:
We can actually avoid setting a version on URL so the reader would always navigate to the latest version.
```suggestion
To read the data from the file, we are going to use the [File](https://api.dart.dev/stable/dart-io/File-class.html) class, providing an ability to get the file content using the following code:
```

```dart
final supportedStorageVersion = File('metrics/STORAGE_VERSION').readAsStringSync();
```

> Once we are able to get the storage version from the file, let's confirm that we can set this version to the `Metrics Web` application environment. Consider the following command that explains the way of adding any environment variables to the `Flutter` application environment during the building process:

Reviewer: :label:
What do you think?
```suggestion
Once we are able to get the storage version from the file, let's clarify that we can set this version to the `Metrics Web` application environment. Consider the following command that explains the way of adding any environment variables to the `Flutter` application environment during the building process:
```
